lib/s3middleware/objectLegalHold.js

```diff
@@ -0,0 +1,112 @@
+const { parseString } = require('xml2js');
+const errors = require('../errors');
+
+/*
+    Format of the xml request:
+    <LegalHold>
+        <Status>ON|OFF</Status>
+    </LegalHold>
+*/
+
+/**
+ * @param {string[]} status - legal hold status parsed from xml to be validated
+ * @return {Error|object} - legal hold status or error
```

Comment:
JSdoc supports type defs or multiple returns for objects, https://jsdoc.app/tags-typedef.html
Might be beneficial here and in other places.
( Won't deny PR though )